src/workerd/api/sql.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/jsg/jsg.h>
+#include "basics.h"
+
+namespace workerd::api {
+
+typedef kj::OneOf<kj::Array<const byte>, kj::String, double> ValueBind;
+class SqlDatabase;
+
+class SqlResult final: public jsg::Object {
+public:
+  SqlResult(SqliteDatabase::Query& query);
+
+  kj::ArrayPtr<jsg::Dict<SqliteDatabase::Query::ValueOwned>> getResults(jsg::Lock& js);
+
+  JSG_RESOURCE_TYPE(SqlResult, CompatibilityFlags::Reader flags) {
+    JSG_READONLY_PROTOTYPE_PROPERTY(results, getResults);
+  }
+
+private:
+  kj::Vector<jsg::Dict<SqliteDatabase::Query::ValueOwned>> results;
+};
+
+class SqlPreparedStatement final: public jsg::Object {
+public:
+  SqlPreparedStatement(jsg::Ref<SqlDatabase>&& sqlDb, SqliteDatabase::Statement&& statement);
+
+  jsg::Ref<SqlResult> run(kj::Maybe<kj::Array<ValueBind>> bindValuesOptional);
+
+  JSG_RESOURCE_TYPE(SqlPreparedStatement, CompatibilityFlags::Reader flags) {
+    JSG_METHOD(run);
+  }
+
+private:
+  void visitForGc(jsg::GcVisitor& visitor) {
+    visitor.visit(sqlDatabase);
+  }
+
+  jsg::Ref<SqlDatabase> sqlDatabase;
+  SqliteDatabase::Statement statement;
+};
+
+class SqlDatabase final: public jsg::Object {
+public:
+  friend class SqlPreparedStatement;
+  SqlDatabase(SqliteDatabase& sqlite);
+
+  jsg::Ref<SqlResult> exec(jsg::Lock& js, kj::String query, bool admin, kj::Maybe<kj::Array<ValueBind>> bindValuesOptional);
+  jsg::Ref<SqlPreparedStatement> prepare(jsg::Lock& js, kj::String query, bool admin);
```

Comment:
Nit: I'm generally not a fan of raw boolean options like this, especially for JS APIs. Can we define an options struct for these?

```js
struct ExecOptions {
  bool admin = false;
  kj::Maybe<kj::Array<ValueBind>> bindValues;
  JSG_STRUCT(admin, bindValues);
};

jsg::Ref<SqlResult> exec(jsg::Lock& js, kj::String query, jsg::Optional<ExecOptions> options);

struct PrepareOptions {
  bool admin = false;
  JSG_STRUCT(admin)
};
jsg::Ref<SqlPreparedStatement> prepare(jsg::Lock& js, kj::String query, jsg::Optional<PrepareOptions> options);
```